shell/platform/android/android_shell_holder.cc

```diff
@@ -34,11 +34,12 @@ AndroidShellHolder::AndroidShellHolder(
   thread_host_ = {thread_label, ThreadHost::Type::UI | ThreadHost::Type::GPU |
                                     ThreadHost::Type::IO};
 
-  // Detach from JNI when the UI thread exits.
-  thread_host_.ui_thread->GetTaskRunner()->PostTask(
-      [key = thread_destruct_key_]() {
-        FXL_CHECK(pthread_setspecific(key, reinterpret_cast<void*>(1)) == 0);
-      });
+  // Detach from JNI when the UI and GPU threads exit.
+  fxl::Closure jni_exit_task([key = thread_destruct_key_]() {
```

Comment:
Nit: You can just say `auto jni_exit_task`. I don't specifying this adds any value and we are going to get rid of fxl anyway.